pkg/operator/controllers/banner/banner.go

```diff
@@ -0,0 +1,52 @@
+package banner
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+
+	consolev1 "github.com/openshift/api/console/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+)
+
+const (
+	BannerName         = "openshift-aro-sre"
+	TextContactSupport = "Please ask your cluster administrator to contact Azure or Red Hat support. Your cluster's resource ID: %s"
+)
+
+func (r *BannerReconciler) reconcileBanner(ctx context.Context, instance *arov1alpha1.Cluster) (err error) {
+	switch instance.Spec.Banner.Content {
+	case arov1alpha1.BannerDisabled:
+		err = r.consolecli.ConsoleV1().ConsoleNotifications().Delete(ctx, BannerName, metav1.DeleteOptions{})
+		if err != nil && kerrors.IsNotFound(err) {
+			//we don't care if the object doesn't exist
+			err = nil
+		}
+	case arov1alpha1.BannerContactSupport:
+		_, err = r.consolecli.ConsoleV1().ConsoleNotifications().Create(ctx, r.newBanner(TextContactSupport, instance.Spec.ResourceID), metav1.CreateOptions{})
+	}
+	return err
+}
+
+func (r *BannerReconciler) newBanner(text string, resourceID string) *consolev1.ConsoleNotification {
+	return &consolev1.ConsoleNotification{
+		TypeMeta: metav1.TypeMeta{
```

Comment:
Do we have to specify TypeMeta?  I thought it was inferred based on the struct you were creating for the given type.

Reply:
Turns out we don't. Removed.

Reply:
It depends on the marshaler you use: if you use `encoding/json` from standard library - you have to explicitly set it. If you use scheme aware one - then you don't.